File: Reporting/CS/ReportMainDemo/AIOperations/AIExceptionHandler.cs
using System;
using DevExpress.AIIntegration;

namespace XtraReportsDemos.AIOperations {
    public class AIExceptionHandler : IAIExceptionHandler {
        public void ProcessException(AIExceptionArgs args) {
            if(!(args.Exception is AIDemoException))
                args.Exception = new Exception("Something went wrong.", args.Exception);
        }
    }
}

File: Reporting/VB/ReportMainDemo/AIOperations/AIExceptionHandler.vb
Imports System
Imports DevExpress.AIIntegration

Namespace XtraReportsDemos.AIOperations

    Public Class AIExceptionHandler
        Implements IAIExceptionHandler

        Public Sub ProcessException(ByVal args As AIExceptionArgs) Implements IAIExceptionHandler.ProcessException
            If Not(TypeOf args.Exception Is AIDemoException) Then args.Exception = New Exception("Something went wrong.", args.Exception)
        End Sub
    End Class
End Namespace
